CONTRIBUTING.md

```diff
@@ -109,42 +115,346 @@ Check out [building instructions](https://github.com/prestodb/presto/tree/master
+               * @see TransformationState#finished()
+               */
+              TransformationState<R> process(Optional<T> elementOptional);
+          }
+      ```
+
+    * Recommend to add the “/** */” style comments to important or difficult public methods, with explanation of the parameters and returns.
+    * Within the method body, use “//” style comments on separate lines for difficult parts.
+    * Use “//” style comments for class members if it can help understanding the code. Can be on the same line or separate lines in front.
+* **Code succinctness**
+    * Inline function calls when appropriate. E.g. if a function is only called once, we don’t need to create a variable for it.
+    * Lambda
+      E.g. Instead of doing this
+
+      ```java
+      Iterables.getOnlyElement(argumentSets).stream()
+      .map((rowExpression) -> OriginalExpressionUtils.castToExpression(rowExpression))
+      ```
+      We just do
+      ```java
+      Iterables.getOnlyElement(argumentSets).stream()
+      .map(OriginalExpressionUtils::castToExpression)
+      ```
+
+* When appropriate use Java 8 Stream API
+* Categorize errors when throwing an exception
+
+
+## Commit Standards
+* ### Commit Size
+    * Recommended lines of code should be within +1000 lines, and should focus on one single major topic.\
+      If your commit is more than 1000 lines, consider breaking it down into multiple commits, each handling a specific small topic.
+* ### Commit Message Style
+    * **Separate subject from body with a blank line**
+    * **Subject**
+        * Limit the subject line to 10 words or 50 characters
+        * If you cannot make the subject short, you may be committing too many changes at once
+        * Capitalize the subject line
+        * Do not end the subject line with a period
+        * Use the imperative mood in the subject line
+    * **Body**
+        * Wrap the body at 72 characters
+        * Use the body to explain what and why vs. how
+        * Use the imperative mood in the body\
+          E.g. “If applied, this commit will ___________”
+        * Communicate only context (why) for the commit in the subject line
+        * Use the body for What and Why, if your commit is complex or dense, share some of the How context
+        * Assume someone may need to revert your change during an emergency
+    * **Content**
+        * **Aim for smaller commits for easier review and simpler code maintenance**
+        * All bug fixes and new features must have associated tests
+        * Commits should pass tests on their own, not be dependent on other commits\
+        * The following is recommended:
+            * Describe why a change is being made.
+            * How does it address the issue?
+            * What effects does the patch have?
+            * Do not assume the reviewer understands what the original problem was.
+            * Do not assume the code is self-evident/self-documenting.
```

Comment:
```suggestion
            * Do not assume the code is self-evident or self-documenting.
```
Avoid slashes. See [GitLab doc recommended word list entry for slashes](https://docs.gitlab.com/ee/development/documentation/styleguide/word_list.html#slashes) for advice and suggestions.